### PR TITLE
Update internal_deps to latest and test with Bazel 6.0.0rc4 as well as Bazel 5.3.2

### PR DESCRIPTION
**Pull request by gregmagolan**
Also adds the GH actions code to cancel previous CI runs for a PR (same as https://github.com/aspect-build/rules_js/pull/694) and to run `test.sh` if that is found at the root of the matrix folder. This is a simple and robust way to run e2e tests that need to make multiple bazel invocations or do other work besides `bazel test ...`

### CODE REVIEW

File: mylang/private/maybe.bzl [hunk 1]
```diff
@@ -0,0 +1,7 @@
+"maybe utilities"
```
Comment (alexeagle) [inline mylang/private/maybe.bzl:1]: would there be more of them? This file doesn't feel worth the indirection to me, the two lines in internal_deps.bzl felt fine

Comment (gregmagolan) [reply mylang/private/maybe.bzl:1]: its a chicken and egg on this one; this little one liner is not something you can load from another rule set but rules almost always will end up with `internal_deps.bzl` and `dependencies.bzl` and it makes both of those much cleaner.

Comment (gregmagolan) [reply mylang/private/maybe.bzl:1]:
```
load("@bazel_tools//tools/build_defs/repo:http.bzl", _http_archive = "http_archive")
load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")

def http_archive(name, **kwargs):
    maybe(_http_archive, name = name, **kwargs)
```
reduces to just one line
```
load("//mylang/private:maybe.bzl", http_archive = "maybe_http_archive")
```
in both those files and any other files a rule set has http_archives deps in

Comment (gregmagolan) [reply mylang/private/maybe.bzl:1]: the proper place would be in @bazel_tools but that may be a hard-one to upstream
```
load("@bazel_tools//tools/build_defs/repo:http.bzl", http_archive = "maybe_http_archive")
```

Comment (gregmagolan) [reply mylang/private/maybe.bzl]: removed from this PR; can do it as a follow-up if it is desirable

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -11,22 +11,52 @@ on:
...
 
+concurrency:
+    # Cancel previous actions from the same PR: https://stackoverflow.com/a/72408109
+    group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+    cancel-in-progress: true
+
 jobs:
+  # matrix-prep-* steps dynamically generate a bit of JSON depending on whether our action has
+  # access to repository secrets. When running on a pull_request from a fork, the author is
+  # untrusted so the secret will be absent. Insanely complex for how simple this requirement is...
```
Comment (alexeagle) [inline .github/workflows/ci.yaml:22]: I think the part about secrets doesn't belong, that's not the reason this uses dynamic matrix

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -11,22 +11,52 @@ on:
...
+  # inspired from
+  # https://stackoverflow.com/questions/65384420/how-to-make-a-github-action-matrix-element-conditional
+
+  matrix-prep-bazelversion:
+    # Prepares the 'bazelversion' axis of the test matrix
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - id: bazel_6
```
Comment (alexeagle) [inline .github/workflows/ci.yaml:31]: nit: `6` will be out-of-date if someone changes the bazelversion file. Maybe it should be `bazel_from_bazelversion`

File: docs/rules.md [hunk 4]
```diff
@@ -1,4 +1,4 @@
 <!-- Generated with Stardoc: http://skydoc.bazel.build -->
 
-Public API re-exports
+
```
Comment (alexeagle) [inline docs/rules.md:3]: that's surprising, why did that line go away?

Comment (gregmagolan) [reply docs/rules.md:3]: stardoc change I'm guessing

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (gregmagolan) [reply .github/workflows/ci.yaml:31]: 👍

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (gregmagolan) [reply .github/workflows/ci.yaml:22]: 👍

File: docs/rules.md (hunk 4 above)
Comment (alexeagle) [reply docs/rules.md:3]: https://github.com/bazel-contrib/rules-template/blob/main/mylang/defs.bzl#L1

the text is still there, so something feels broken here.

Comment (gregmagolan) [reply docs/rules.md:3]: ahh. if there are no exported symbols the generated file is empty now

Comment (gregmagolan) [reply docs/rules.md:15]: When I add
```
def example():
  """This is an example"""
  print("example export")
```
Then the .md is no longer empty

**Review by alexeagle — APPROVED** [review]

**Pull request merged**